sdk/cosmos/azure-cosmos/azure/cosmos/aio/_retry_utility_async.py

```diff
@@ -194,6 +195,15 @@ async def send(self, request):
                 timeout_error.response = response
                 timeout_error.history = retry_settings['history']
                 raise
+            except ServiceRequestError as err:
+                # the request ran into a socket timeout or failed to establish a new connection
+                # since request wasn't sent, we retry up to however many connection retries are configured (default 3)
+                if retry_settings['connect'] > 0:
```

Comment:
Doesn't your pipeline already have a retry policy? If so, that will also be retrying already.

Reply:
I'm not sure I understand the question, this is our connection retry policy that gets sent to our pipeline on client creation: 
![image](https://github.com/user-attachments/assets/69e8f47f-01dc-49db-8742-5ceeacc7e00f)

![image](https://github.com/user-attachments/assets/c16bf61c-6c09-4857-8612-9dcfd438deeb)

Reply:
Ahhh sorry - I should have expanded the diff to see the class properly before commenting :) Ignore me!